examples/thermostat/linux/thermostat-delegate-impl.cpp

```diff
@@ -148,6 +148,44 @@ CHIP_ERROR ThermostatDelegate::SetActivePresetHandle(const DataModel::Nullable<B
     return CHIP_NO_ERROR;
 }
 
+System::Clock::Milliseconds16
+ThermostatDelegate::GetAtomicWriteTimeout(DataModel::DecodableList<chip::AttributeId> attributeRequests,
+                                          System::Clock::Milliseconds16 timeoutRequest)
+{
+    auto attributeIdsIter = attributeRequests.begin();
+    bool requestedPresets = false, requestedSchedules = false;
+    while (attributeIdsIter.Next())
+    {
+        auto & attributeId = attributeIdsIter.GetValue();
+
+        switch (attributeId)
+        {
+        case Attributes::Presets::Id:
+            requestedPresets = true;
+            break;
+        case Attributes::Schedules::Id:
+            requestedSchedules = true;
+            break;
+        default:
+            return System::Clock::Milliseconds16(0);
+        }
+    }
+    if (attributeIdsIter.GetStatus() != CHIP_NO_ERROR)
+    {
+        return System::Clock::Milliseconds16(0);
+    }
+    auto timeout = System::Clock::Milliseconds16(0);
+    if (requestedPresets)
+    {
+        timeout += std::chrono::milliseconds(1000);
+    }
+    if (requestedSchedules)
+    {
+        timeout += std::chrono::milliseconds(3000);
```

Comment:
This could use some documentation explaining where these numbers are coming from and what they mean.

Also:

1. I suspect in practice on a somewhat-congested Thread network those numbers might be low.  We should put numbers here that will work ok if someone just blindly copies this whole delegate implementation.
2. It would be good to have some documentation at the top of the method explaining what this method _has_ to do vs what we are just doing as an example of things we _could_ do.